<?xml version="1.0" ?>


<!-- Seed: 3759675 -->
<sdf version="1.6">
  <world name="ServiceSim">

    <!-- GUI -->
    <gui fullscreen='0'>
      <camera name='user_camera'>
        <!--pose>-14.41 -19.9 4.14 0 0.5 0.76</pose-->
        <pose>25.9 -21.8 73.6 0 1.05 1.57</pose>
      </camera>
    </gui>

    <!-- Scene -->
    <scene>
      <ambient>0.8 0.8 0.8 1.0</ambient>
      <background>0.66 0.77 0.84 1.0</background>
      <grid>1</grid>
      <origin_visual>1</origin_visual>
    </scene>

    <!-- A global light source -->
<!--
    <light type="directional" name="sun">
      <cast_shadows>true</cast_shadows>
      <pose>0 0 1000 0 0 0</pose>
      <diffuse>0.8 0.8 0.8 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>
-->

    <!-- Ground collision -->
    <model name="ground_collision">
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>80 80</size>
            </plane>
          </geometry>
        </collision>
      </link>
    </model>

    <!-- Office -->
    
    

<model name="reception">

  <static>true</static>

  <pose>
    4.5
    0.0
    0.0
    0.0
    0.0
    3.141592653589793
  </pose>

  <link name="link">
    <collision name="collision">
      <geometry>
        <mesh>
          <uri>file://media/meshes/reception.obj</uri>
          <scale>0.01 0.01 0.01</scale>
        </mesh>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh>
          <uri>file://media/meshes/reception.obj</uri>
          <scale>0.01 0.01 0.01</scale>
        </mesh>
      </geometry>
    </visual>
  </link>

</model>


    
    

<model name="meeting_room_large">

  <static>true</static>

  <pose>
    6.0
    13.6
    0.0
    0.0
    0.0
    -1.5707963267948966
  </pose>

  <link name="link">
    <collision name="collision">
      <geometry>
        <mesh>
          <uri>file://media/meshes/meeting_room_large.obj</uri>
          <scale>0.01 0.01 0.01</scale>
        </mesh>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh>
          <uri>file://media/meshes/meeting_room_large.obj</uri>
          <scale>0.01 0.01 0.01</scale>
        </mesh>
      </geometry>
    </visual>
  </link>

</model>


    
    

<model name="meeting_room_small">

  <static>true</static>

  <pose>
    6.0
    19.7
    0.0
    0.0
    0.0
    -1.5707963267948966
  </pose>

  <link name="link">
    <collision name="collision">
      <geometry>
        <mesh>
          <uri>file://media/meshes/meeting_room_small.obj</uri>
          <scale>0.01 0.01 0.01</scale>
        </mesh>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh>
          <uri>file://media/meshes/meeting_room_small.obj</uri>
          <scale>0.01 0.01 0.01</scale>
        </mesh>
      </geometry>
    </visual>
  </link>

</model>


    
    

<model name="refreshment_area">

  <static>true</static>

  <pose>
    16.6
    7.5
    0.0
    0.0
    0.0
    0.0
  </pose>

  <link name="link">
    <collision name="collision">
      <geometry>
        <mesh>
          <uri>file://media/meshes/refreshment_area.obj</uri>
          <scale>0.01 0.01 0.01</scale>
        </mesh>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh>
          <uri>file://media/meshes/refreshment_area.obj</uri>
          <scale>0.01 0.01 0.01</scale>
        </mesh>
      </geometry>
    </visual>
  </link>

</model>


    
    

<model name="office_large">

  <static>true</static>

  <pose>
    25.5
    3.0
    0.0
    0.0
    0.0
    1.5707963267948966
  </pose>

  <link name="link">
    <collision name="collision">
      <geometry>
        <mesh>
          <uri>file://media/meshes/office_large.obj</uri>
          <scale>0.01 0.01 0.01</scale>
        </mesh>
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh>
          <uri>file://media/meshes/office_large.obj</uri>
          <scale>0.01 0.01 0.01</scale>
        </mesh>
      </geometry>
    </visual>
  </link>

</model>


<!--
    <model name="office">
      <static>true</static>

      
      <link name="office_small">
        <pose>
          3.0
          45.1
          0
          0
          0
          -1.5707963267948966
        </pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/office_small.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/office_small.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="bathroom">
        <pose>10 0 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/bathroom.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/bathroom.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="carpet">
        <pose>20 0 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/carpet.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/carpet.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="ceiling">
        <pose>30 0 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/ceiling.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/ceiling.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="hallway">
        <pose>10 10 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/hallway.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/hallway.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="meeting_room_large">
        <pose>10 20 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/meeting_room_large.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/meeting_room_large.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="meeting_room_small">
        <pose>10 30 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/meeting_room_small.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/meeting_room_small.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="refreshment_area">
        <pose>10 40 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/refreshment_area.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/refreshment_area.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="sink">
        <pose>20 10 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/sink.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/sink.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="tile">
        <pose>20 30 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/tile.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/tile.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="office_large">
        <pose>30 20 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/office_large.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/office_large.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="wall_door">
        <pose>40 10 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/wall_door.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/wall_door.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="wall_elevator">
        <pose>40 20 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/wall_elevator.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/wall_elevator.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="wall_plain">
        <pose>40 30 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/wall_plain.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/wall_plain.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

      <link name="wall_window">
        <pose>40 40 0 0 0 0</pose>
        <collision name="collision">
          <geometry>
            <mesh>
              <uri>file://media/meshes/wall_window.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <mesh>
              <uri>file://media/meshes/wall_window.obj</uri>
              <scale>0.01 0.01 0.01</scale>
            </mesh>
          </geometry>
        </visual>
      </link>

    </model>
-->
    <!-- Trajectory actors -->
    
      

<actor name="actor_1">

  <!-- Starting pose, nice for when the world is reset -->
  <pose>
    -9.32582
    -15.9673
    1.2138
    1.57
    0.0
    -0.22373
  </pose>

  <skin>
    <filename>model://actor/meshes/SKIN_man_red_shirt.dae</filename>
  </skin>

  <animation name="animation">
    <filename>model://actor/meshes/ANIMATION_running.dae</filename>
    <interpolate_x>true</interpolate_x>
  </animation>

  <plugin name="trajectory" filename="libTrajectoryActorPlugin.so">
    
      <target>
        -9.32582
        -15.9673
        1.2138
        1.57
        0.0
        -0.22373
      </target>
    
      <target>
        -9.201666
        -7.601102
        1.2138
        1.57
        0.0
        -0.22373
      </target>
    
      <target>
        5.87782
        -7.59315
        1.2138
        1.57
        0.0
        -0.22373
      </target>
    
      <target>
        13.2181
        -10.6971
        1.2138
        1.57
        0.0
        -0.22373
      </target>
    
      <target>
        15.2346
        -13.7236
        1.2138
        1.57
        0.0
        -0.22373
      </target>
    
      <target>
        15.2558
        -16.0176
        1.2138
        1.57
        0.0
        -0.22373
      </target>
    
      <target>
        -9.32582
        -15.9673
        1.2138
        1.57
        0.0
        -0.22373
      </target>
    

    <velocity>1.5</velocity>
    <ignore_obstacle>willowgarage</ignore_obstacle>
    <ignore_obstacle>ground_collision</ignore_obstacle>

  </plugin>

</actor>

    
      

<actor name="actor_2">

  <!-- Starting pose, nice for when the world is reset -->
  <pose>
    13.4107
    8.18946
    1.2138
    1.570796
    -0.0
    3.141593
  </pose>

  <skin>
    <filename>model://actor/meshes/SKIN_man_red_shirt.dae</filename>
  </skin>

  <animation name="animation">
    <filename>model://actor/meshes/ANIMATION_walking.dae</filename>
    <interpolate_x>true</interpolate_x>
  </animation>

  <plugin name="trajectory" filename="libTrajectoryActorPlugin.so">
    
      <target>
        13.4107
        8.18946
        1.2138
        1.570796
        -0.0
        3.141593
      </target>
    
      <target>
        12.9881
        19.467501
        1.2138
        1.570796
        -0.0
        3.141593
      </target>
    
      <target>
        35.577293
        19.779964
        1.2138
        1.570796
        -0.0
        3.141593
      </target>
    
      <target>
        35.539101
        17.879601
        1.2138
        1.570796
        -0.0
        3.141593
      </target>
    
      <target>
        27.718399
        17.094101
        1.2138
        1.570796
        -0.0
        3.141593
      </target>
    
      <target>
        28.0737
        8.25699
        1.2138
        1.570796
        -0.0
        3.141593
      </target>
    

    <velocity>1.1161902237033754</velocity>
    <ignore_obstacle>willowgarage</ignore_obstacle>
    <ignore_obstacle>ground_collision</ignore_obstacle>

  </plugin>

</actor>

    

    <!-- Idling actors -->
    
      

<actor name="actor_3">

  <!-- Starting pose, nice for when the world is reset -->
  <pose>
    -11.969092
    -17.137756
    0.15
    0.0
    -0.0
    0.719669
  </pose>

  <skin>
    <filename>model://actor/meshes/SKIN_man_blue_shirt.dae</filename>
  </skin>

  <animation name="animation">
    <filename>model://actor/meshes/ANIMATION_talking_b.dae</filename>
  </animation>

  <!-- Need one waypoint to idle at -->
  <script>
    <trajectory id='0' type='animation'>
      <waypoint>
        <time>100</time>
          <pose>
            -11.969092
            -17.137756
            0.15
            0.0
            -0.0
            0.719669
          </pose>
      </waypoint>
    </trajectory>
  </script>

</actor>

    
      

<actor name="actor_4">

  <!-- Starting pose, nice for when the world is reset -->
  <pose>
    6.323622
    -6.638186
    0.15
    0.0
    0.0
    -0.22373
  </pose>

  <skin>
    <filename>model://actor/meshes/SKIN_man_blue_shirt.dae</filename>
  </skin>

  <animation name="animation">
    <filename>model://actor/meshes/ANIMATION_talking_b.dae</filename>
  </animation>

  <!-- Need one waypoint to idle at -->
  <script>
    <trajectory id='0' type='animation'>
      <waypoint>
        <time>100</time>
          <pose>
            6.323622
            -6.638186
            0.15
            0.0
            0.0
            -0.22373
          </pose>
      </waypoint>
    </trajectory>
  </script>

</actor>

    

  </world>
</sdf>
